tests/test-class-amp-validation-utils.php

```diff
@@ -131,6 +131,89 @@ public function test_add_validation_hooks() {
 		$this->assertEquals( -1, has_action( 'do_shortcode_tag', array( self::TESTED_CLASS, 'decorate_shortcode_source' ) ) );
 	}
 
+	/**
+	 * Test add_validation_hooks with Gutenberg active.
+	 *
+	 * @covers AMP_Validation_Utils::add_validation_hooks()
+	 */
+	public function test_add_validation_hooks_gutenberg() {
+		if ( ! function_exists( 'do_blocks' ) ) {
+			$this->markTestSkipped( 'Gutenberg not active.' );
+		}
+
+		$this->assertEquals( 9, has_filter( 'the_content', 'do_blocks' ) );
+		AMP_Validation_Utils::add_validation_hooks();
+		$this->assertFalse( has_filter( 'the_content', 'do_blocks' ) );
+	}
+
+	/**
+	 * Get block data.
+	 *
+	 * @ses Test_AMP_Validation_Utils::test_do_blocks()
+	 * @return array
+	 */
+	public function get_block_data() {
+		return array(
+			'paragraph'    => array(
+				"<!-- wp:paragraph -->\n<p>Latest posts:</p>\n<!-- /wp:paragraph -->",
+				"<!--amp-source-stack {\"block_name\":\"core\/paragraph\"}-->\n<p>Latest posts:</p>\n<!--/amp-source-stack {\"block_name\":\"core\/paragraph\"}-->",
+				array(
+					'element' => 'p',
+					'blocks'  => array( 'core/paragraph' ),
+				),
+			),
+			'latest_posts' => array(
+				'<!-- wp:latest-posts /-->',
+				'<!--amp-source-stack {"block_name":"core\/latest-posts","type":"plugin","name":"gutenberg","function":"render_block_core_latest_posts"}--><ul class="wp-block-latest-posts aligncenter"></ul><!--/amp-source-stack {"block_name":"core\/latest-posts","type":"plugin","name":"gutenberg","function":"render_block_core_latest_posts"}-->',
+				array(
+					'element' => 'ul',
+					'blocks'  => array( 'core/latest-posts' ),
+				),
+			),
+			'columns'      => array(
+				"<!-- wp:columns -->\n<div class=\"wp-block-columns has-2-columns\">\n    <!-- wp:quote {\"layout\":\"column-1\",\"foo\":{\"bar\":1}} -->\n    <blockquote class=\"wp-block-quote layout-column-1\">\n        <p>A quotation!</p><cite>Famous</cite></blockquote>\n    <!-- /wp:quote -->\n\n    <!-- wp:html {\"layout\":\"column-2\"} -->\n    <div class=\"layout-column-2\">\n        <script>\n            document.write('Not allowed!');\n        </script>\n    </div>\n    <!-- /wp:html -->\n</div>\n<!-- /wp:columns -->",
```

Comment:
Gutenberg creates blocks with `wp:columns`: https://github.com/Automattic/amp-wp/pull/1019/files#diff-ae6e35f6dd94607ca0bb8954533a1cb5R174

But they are synonymous. The `core/` is implict when no namespace is present: https://github.com/WordPress/gutenberg/blob/11187b4093df792f4a5caefc765acca98d5fbb04/lib/blocks.php#L163-L164